fix: handle empty overrideScorers in agent scoring

YujohnNattrass opened the pull request:
When `overrideScorers` was passed as an empty object `{}`, the agent would throw a "No scorers found" error because `{}` is truthy but has no entries.

Now:
- `overrideScorers: undefined` → uses default scorers via `listScorers()`
- `overrideScorers: {}` → early return, skips scoring entirely  
- `overrideScorers: { ... }` → resolves and uses the provided scorers

```typescript
// Before: throws error
await agent.generate({ overrideScorers: {} })

// After: skips scoring gracefully
await agent.generate({ overrideScorers: {} })
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed scorer override handling to gracefully skip scoring when provided with an empty override configuration.
  * Enhanced error handling for cases where specified override scorers cannot be resolved, with improved error messaging.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

YujohnNattrass committed:
fix: handle empty overrideScorers in agent scoring

abhiaiyer91 commented on packages/core/src/agent/agent.ts:
```diff
@@ -2412,6 +2412,11 @@ export class Agent<TAgentId extends string = string, TTools extends ToolsInput =
     tracingContext: TracingContext;
   }) {
+    // If overrideScorers is explicitly empty, skip scoring
+    if (overrideScorers && Object.keys(overrideScorers).length === 0) {
```
`overrideScorers || {}`

YujohnNattrass committed:
refactor(core): handle empty overrideScorers in resolveOverrideScorerReferences

Move the empty overrideScorers check into resolveOverrideScorerReferences
instead of doing an early return. The method now only throws if scorers
were provided but none could be resolved, allowing empty {} to pass
through gracefully.
